pkg/codegen/dotnet/gen.go

```diff
@@ -652,6 +681,25 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 		fmt.Fprintf(w, "        }\n")
 	}
 
+	if hasConstInputs {
+		// Write the method that will calculate the resource arguments.
+		fmt.Fprintf(w, "\n")
+		fmt.Fprintf(w, "        private static %[1]s MakeArgs(%[1]s args)\n", argsType)
+		fmt.Fprintf(w, "        {\n")
+		fmt.Fprintf(w, "            args ??= new %sArgs();\n", className)
+		for _, prop := range r.InputProperties {
+			if prop.ConstValue != nil {
+				v, err := mod.getConstValue(prop.ConstValue)
+				if err != nil {
+					return err
+				}
+				fmt.Fprintf(w, "            args.%s = %s;\n", mod.propertyName(prop), v)
+			}
```

Comment:
If we're going to do this `nil` check here, we might as well just inline `getConstValue` and remove that method:

```suggestion
			if prop.ConstValue != nil {
				v, err := primitiveValue(prop.ConstValue)
				if err != nil {
					return err
				}
				fmt.Fprintf(w, "            args.%s = %s;\n", mod.propertyName(prop), v)
			}
```